Track `@types/*` packages

mxdvl opened the pull request:
<!-- In this repo you can label a PR with the "PR Deployment" label to deploy the code to a publicly accessible url -->
## What does this change?

Create a regularly updating issue (#6965) that tracks all `major.minor` versions for packages with a matching `@types/*`.

## Why?

Mismatches in @types packages significantly reduce any type safety gained by using TypeScript.

## Screenshots

<img width="365" alt="image" src="https://user-images.githubusercontent.com/76776/212701614-38395946-6b22-4e89-8dde-4c42eb242481.png">

jamesgorrie commented on scripts/deno/types-dependencies.ts:
```diff
@@ -0,0 +1,80 @@
+		[dependency, version.match(/(\d+\.\d+)/)?.at(0)] as const,
+);
+
+// console.log(majorMinorVersions);
```
👀 ?

jamesgorrie reviewed (commented):
How do we see this working in terms of telling us when things are wrong? The comment is really useful, but what could be really _really_ useful is having alerting for when we regress, or even a weekly rundown?

joecowton1 commented:
> How do we see this working in terms of telling us when things are wrong? The comment is really useful, but what could be really _really_ useful is having alerting for when we regress, or even a weekly rundown?

Agree, also really needs an investment of time to resolve some of the wider issues at play here - visibility is great but it doesn't fix anything.

I think when everyone is ready to upgrade to React 18 there will scope to eliminate at least a few of the mismatches.

mxdvl commented:
> How do we see this working in terms of telling us when things are wrong? The comment is really useful, but what could be really really useful is having alerting for when we regress, or even a weekly rundown?

> Agree, also really needs an investment of time to resolve some of the wider issues at play here - visibility is great but it doesn't fix anything.

I’ve now merged the tilde range CI action from #6971, @joecowton1 & @jamesgorrie, which addresses these concerns.

bryophyta commented:
@mxdvl is this a PR that you want to merge into main before #7003? Or do you want to merge #7003 into this branch before merging this to main?

mxdvl commented:
> @mxdvl is this a PR that you want to merge into main before #7003? Or are you intending to merge #7003 into this branch before merging this to main?

I’m thinking that this can be superceded by #7100

bryophyta commented:
> I’m thinking that this can be superceded by https://github.com/guardian/dotcom-rendering/pull/7100

Okay thanks! In that case I'll add the `prnouncer-ignore` label to this PR for the time being, but obv feel free to remove it if you do want this PR to be reviewed!
